Sources/ProjectDescription/TargetDependency.swift

```diff
@@ -47,29 +47,29 @@ public enum TargetDependency: Codable, Hashable {
     ///
     /// - Parameters:
     ///   - name: Name of the target to depend on
-    case target(name: String)
+    case target(name: String, platformFilters: PlatformFilters = .all)
```

Comment:
I think it's strange if all cases require the same associated values, which I also expect to be relatively rarely used 🤔 

Not 100% sure, but maybe we can do something like:
```
Target(
  …,
  dependencies: [
    .target('A'),
    .target('B').only(on: .iOS),
  ],
)
```
or something similar?

Reply:
what concrete type would that resolve to within that array? these being defined as `enum` types constrain our design space a bit.  i think we could add some instance methods to these to provide some syntactic sugar like that but in the end given we're maintaining source compatibility, I wasnt able to see any other ways to achieve what we need without adding this new associated value across all cases.  

This applies both to the API we expose to users but also how packages are mapped into `ProjectDescription` types during `fetch`

Reply:
x-posting some thoughts on platform filters which are also applicable here https://github.com/tuist/tuist/pull/5557#pullrequestreview-1745191008

Reply:
> think it's strange if all cases require the same associated values, which I also expect to be relatively rarely used 🤔

👍 anecdotally from our project (a graph with 1000s of edges) only 3 of them have filters specified - which may ago at some point too

Reply:
>  I wasnt able to see any other ways to achieve what we need without adding this new associated value across all cases.

Agreed the current approach of adding `platformFilters` is the easiest path to Tuist 3 source compatibility